src/xsdba/nbutils.py

```diff
@@ -181,17 +181,19 @@ def vecquantiles(da: DataArray, rnk: DataArray, dim: str | Sequence[Hashable]) -
     xarray.DataArray
         The quantiles computed along the `dim` dimension.
     """
-    tem = utils.get_temp_dimname(da.dims, "temporal")
     dims = [dim] if isinstance(dim, str) else dim
+    common_dims = list(set(da.dims).intersection(set(rnk.dims)))
+    tem = utils.get_temp_dimname(da.dims, "temporal")
     da = da.stack({tem: dims})
-    da = da.transpose(*rnk.dims, tem)
-
-    res = DataArray(
-        _vecquantiles(da.values, rnk.values),
-        dims=rnk.dims,
-        coords=rnk.coords,
-        attrs=da.attrs,
-    ).astype(da.dtype)
+
+    res = apply_ufunc(
+        _vecquantiles,
+        da,
+        rnk,
+        input_core_dims=[common_dims + [tem], common_dims],
+        output_core_dims=[common_dims],
+        dask="parallelized",
+    )
```

Comment:
only adapt_freq uses this. 

I assume there was no particular reason to use bare numpy instead of through apply_ufunc? 

My assumption is that: 
`da: (dims_to_reduce, common_dims_with_rnk, passenger_dims)`

It was necessary for adapt_freq with ADD_DIMS, which are the passenger dims in this case.